capsules/src/qdec.rs

```diff
@@ -0,0 +1,108 @@
+//! Provides userspace access to the Qdec on a board.
+//!
+//! Usage
+//! -----
+//!
+//! ````
+//! let qdec = static_init!(
+//!     capsules::qdec::Qdec<'static>,
+//!     capsules::qdec::QdecInterface::new(&nrf52::qdec::QDEC,
+//!                                         kernel::Grant::create())
+//! );
+//! kernel::hil::QdecDriver.set_client(qdec);
+//! ````
+//! #Interrupt Spurred Readings versus Regular Readings
+//! An application can either enable interrupts to get the
+//! accumulation value or manually read it whenever it wants
```

Comment:
This reads as something very specific to the nRF implementation. At this layer, however a capsule is ideally a more generic interface to something providing quadrature decoding. It would be good to find one or two other (non-Nordic) chips and quickly verify that their hardware engines can support this interface (no need to implement, just a datasheet read).

Reply:
As of right now, I have looked at the SAM 3/4 chip and it seem its QDEC on board does behave differently than the nRF. The datasheet for that chip's QDEC can be found [here](http://ww1.microchip.com/downloads/en/AppNotes/Atmel-42706-Quadrature-Decoder-QDEC-for-SAM3-4-Devices_ApplicationNote_AT11483.pdf). I'm not sure how to proceed from this-I'm currently looking at other chips to see if it is the same situation for them (particularly chips that support tockOS already). Right now, it seems most don't have a QDEC.